$OpenBSD: patch-src_imlib_c,v 1.11 2012/03/09 08:48:42 dcoppa Exp $

fix memory leak in feh_edit_inplace_lossless
(upstream git commit d0721373fa0a9d189ea95ec736f661cc0df436cd)

--- src/imlib.c.orig	Fri Mar  9 09:36:54 2012
+++ src/imlib.c	Fri Mar  9 09:38:15 2012
@@ -1108,7 +1108,7 @@ void feh_edit_inplace_lossless(winwidget w, int op)
 				"-outfile", file_str, file_str, NULL);
 
 		im_weprintf(w, "lossless %s: Is 'jpegtran' installed? Failed to exec:", op_name);
-		return;
+		exit(1);
 	} else {
 		waitpid(pid, &status, 0);
 
@@ -1118,9 +1118,11 @@ void feh_edit_inplace_lossless(winwidget w, int op)
 					" Commandline was: "
 					"jpegtran -copy all %s %s -outfile %s %s",
 					op_name, status >> 8, op_op, op_value, file_str, file_str);
+			free(file_str);
 			return;
 		}
 	}
+	free(file_str);
 }
 
 void feh_draw_actions(winwidget w)
